server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -1238,6 +1239,28 @@ private void executePipelines(
                         return; // document failed!
                     }
 
+                    for (StreamType streamType : StreamType.getEnabledStreamTypesForProject(project)) {
+                        if (streamType.matchesStreamPrefix(newIndex)
+                            && ingestDocument.getIndexHistory().contains(streamType.getStreamName()) == false) {
+                            exceptionHandler.accept(
+                                new IngestPipelineException(
+                                    pipelineId,
+                                    new IllegalArgumentException(
```

Comment:
The other exceptions like this are all IllegalStateException, should we follow suit?